Heejung/Lv2/괄호 변환.js

```diff
@@ -0,0 +1,42 @@
+// u, v로 쪼개기
+function divide(p) {
+  let u = '';
+  let v = '';
+  let isCorrect = true; // 올바른 괄호인지
+
+  for (let i = 0, tmp = 0; i < p.length; i += 1) {
+    tmp += p[i] === '(' ? 1 : -1;
+    if (tmp < 0) isCorrect = false;
+    if (tmp === 0) {
+      u = p.substring(0, i + 1);
+      v = p.substring(i + 1);
+      break;
+    }
+  }
+  return [u, v, isCorrect];
+}
+
+function solution(p) {
+  // 빈문자열은 빈문자열로 반환
+  if (p === '') return p;
+
+  let [u, v, isCorrect] = divide(p);
+
+  // 잘못된 괄호
+  if (!isCorrect) {
+    return (
+      '(' +
+      solution(v) +
+      ')' +
+      u
+        .slice(1, -1)
+        .split('')
+        .map((str) => (str === '(' ? ')' : '('))
+        .join('')
+    );
+
+    // 올바른 괄호
+  } else {
+    return u + solution(v);
+  }

```

Comment:
```suggestion
  }
  return u + solution(v);

```
if문 내에서 무조건적으로 return이 된다면 else라는 키워드를 없이 코드를 작성해도 결과가 같고 들여쓰기의 깊이가 줄어들기 때문에 가독성이 좋아져요.
no-else-return: https://eslint.org/docs/rules/no-else-return